source/exercises/mailroom-oo.rst

```diff
@@ -51,29 +52,64 @@ One of the hardest parts of OO design (particularly in Python) is to know how "l
+
+Below are more detailed suggestions on breaking down your existing code into multiple modules that will be part of a single mailroom program.
+
+Donor Class
+^^^^^^^^^^^^
+
+Module responsible for donor data encapsulation.
 
-You'll want a Donor class -- this will hold all the information about the donor, and have attributes, properties, and methods to provide access to the donor specific information that is needed.
+This will hold all the information about a single donor, and have attributes, properties, and methods to provide access to the donor specific information that is needed.
+Remember that if you are writing code that only accesses information about a single donor then it should most likely live in this class.
 
-You'll then want a class that handles the collection of donors. This will hold all the donor objects, but also methods to add a new donor, search for a given donor, etc. If you want a way to save and re-load your data, this class would have that too.
+DonorCollection Class
+^^^^^^^^^^^^^^^^^^^^^
+
+Module responsible for donor collection data encapsulation.
+
+This will hold all of the donor objects, but also methods to add a new donor, search for a given donor, etc. If you want a way to save and re-load your data, this class would have that too.
 
 Your class for the collection of donors will also hold the code that generates reports about multiple donors.
 
-Remember that the user interaction code (anything with an ``input`` function) should be outside of these "logic" or "model" classes.
 
-In general, you will have a method for each of the functions in your non-OO version. Which class they go it will depend on whether the method only needs the information from one donor, or from the whole collection.
+Command Line Interface
+^^^^^^^^^^^^^^^^^^^^^^
+
+Module responsible for main program flow (CLI - Command Line Interface).
+
+Let's call this module ``cli_main.py`` to represent the entry point for the mailroom program. This module will be using above classes we defined: ``Donor`` and ``DonorCollection``; it will also handle interaction with the user via the ``input`` function calls to gather user input and to provide the output to the console.
+
+What should go into this module?
+
+* the main "switch dictionary" to map user selection to the program features; in general, you will have a method for each of the mailroom functions.
+* ``input`` function calls to gather user input.
+* ``print`` statements to print to console.
+
+
+.. note::  Technically console print statement don't belong in your data classes, however for some features like send letters instead of "sending" we are simply printing so it is ok for this feature to reside in the data class.
```

Comment:
I try to have them have all the methods in the data classes return strings, rather than call print() directly.

But yes, at _some_point, you need to print -- but that could be in the CLI or Data classes.